vault/provider_test.go

```diff
@@ -640,3 +600,206 @@ func testTokenName_check(expectedTokenName string) resource.TestCheckFunc {
 		return nil
 	}
 }
+
+// A token helper script that echos back the VAULT_ADDR value
+const echoBackTokenHelperScript = `#!/usr/bin/env bash
+printenv VAULT_ADDR
+`
+
+func TestAccProviderVaultAddrEnv(t *testing.T) {
+	// This is an acceptance test because it requires filesystem and env var
+	// changes that could interfere with other Vault operations.
+	if os.Getenv(resource.TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf(
+			"Acceptance tests skipped unless env '%s' set",
+			resource.TestEnvVar))
+	}
+
+	// Clear the config file env var and restore it after the test.
+	resetConfigPathEnv, err := tempUnsetenv(config.ConfigPathEnv)
+	defer failIfErr(t, resetConfigPathEnv)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// clear BASH_ENV for this test so any cmd.Exec invocations do not source the BASH_ENV
+	// file which is configured with a VAULT_ADDR here:
+	// https://github.com/terraform-providers/terraform-provider-vault/blob/f42716aae3aebc8daf9702dfa20ce3f8d09d9f4d/.circleci/config.yml#L27
+	// All values set in that BASH_ENV file will still be in the process environment of this
+	// test, they just won't clobber any values this test modifies in the process environment
+	// when the ExternalTokenHelper's command is formatted into a shell invocation to exec here:
+	// https://github.com/hashicorp/vault/blob/master/command/token/helper_external.go#L117-L132
+	//
+	resetBashEnvEnv, err := tempUnsetenv("BASH_ENV")
```

Comment:
I think its possible that a better fix here might be to move all of the vars set here: https://github.com/terraform-providers/terraform-provider-vault/blob/f42716aae3aebc8daf9702dfa20ce3f8d09d9f4d/.circleci/config.yml#L26-L32 to an `Environment` object in the circleci yaml configuration for the container these tests are run in, as documented here: https://circleci.com/docs/2.0/env-vars/#setting-an-environment-variable-in-a-container

That being said, I don't know the finer details of these tests or how circle manages those environment variables/containers across multiple steps well enough to be confident that that wouldn't have any unintended side effects, so I took a more surgical route to fixing these tests that rely on testing the VAULT_ADDR change when a token helper is called.